lib/routes/validators.js

```diff
@@ -134,40 +134,67 @@ module.exports.isValidEmailAddress = function(value) {
 }
 
 module.exports.redirectTo = function redirectTo(base) {
-  const redirectTo = isA.string().max(512)
+  const validator = isA.string().max(512)
   let hostnameRegex = null
   if (base) {
     hostnameRegex = new RegExp('(?:\\.|^)' + base.replace('.', '\\.') + '$')
   }
-  redirectTo._tests.push(
+  validator._tests.push(
     {
       func: (value, state, options) => {
         if (value !== undefined && value !== null) {
-          const normalizedValue = module.exports.isValidUrl(value, hostnameRegex)
-          if (normalizedValue) {
-            return normalizedValue
+          if (module.exports.isValidUrl(value, hostnameRegex)) {
+            return value
           }
         }
 
-        return redirectTo.createError('string.base', { value }, state, options)
+        return validator.createError('string.base', { value }, state, options)
       }
     }
   )
-  return redirectTo
+  return validator
 }
 
-module.exports.isValidUrl = function (redirect, hostnameRegex) {
-  var parsed = url.parse(redirect)
+module.exports.url = function url(options) {
+  const validator = isA.string().max(512).uri(options)
+  validator._tests.push(
+    {
+      func: (value, state, options) => {
+        if (value !== undefined && value !== null) {
+          if (module.exports.isValidUrl(value)) {
+            return value
+          }
+        }
+
+        return validator.createError('string.base', { value }, state, options)
+      }
+    }
+  )
+  return validator
+}
+
+module.exports.isValidUrl = function (url, hostnameRegex) {
```

Comment:
There's at least 2 places in this module where this is dereferenced internally as `module.exports.isValidUrl`. If we named the function expression we could make those read nicer as `isValidUrl` or whatever.

(I realise this function already existed but as it's changed I figured it's fair game for comments)

Reply:
Actually, when I grep locally, this isn't even used outside the module. Do we need to export it?

One of the things that was a little bit confusing when I first read this diff was the dichotomy between the similarily-named `url` and `isValidUrl` exports. But if `isValidUrl` is just an internal helper for this module, that makes more sense to me, and I think it would be clearer if it was only callable locally.

Reply:
Good call, let's make it an internal helper.